packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -967,6 +972,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             metadata,
         );
 
+        if (this.garbageCollector.shouldRunSweep) {
```

Comment:
And same concern RE shouldRunSweep: When I look at garbageCollection.ts, like 293, where we calculate this.shouldRunSweep, it's pretty clear that (at least due to clients being able to overwrite settings locally), some clients may enable sweep, and some may not - in a single session. Thus, some clients may have this timer running and some not. Well, if some clients run forever, while other clients run summarizer and delete stuff, then we get data corruption / unpredictable behavior. Thus, I think timer has to be non-conditional.

Reply:
I suggested shouldRunSweep because we don't want to enable this right away (at least until we hear back from the partners and they agree on this). We would want to ship this dark and enable is over time. The config manager system that Tony / Andrei are building will help but until then we need some way to only enable it for certain documents. Should we add a local storage flag for now instead?

The timer can be based off of `gcEnabled` which remains the same throughout the lifetime of a file. Or we may want to do it unconditionally. I don't have a preference either way.

Reply:
Sure, but how will you enable it later? I.e. at some point you will enable sweep, and new clients will use timer. But old clients will not. We need a reliable system that instructs all clients in a session (old and new) about exactly same GC settings. And thus, I think it pushes us to write these settings into a file. This allows us to ship it dark (all "old" files will never have GC, unless there is some "conversion" process, like code proposal that lifts all the boats and at the same time overwrites metafile settings) and enable (for newly created files) in the future.

The only other alternative I see - some kind of consensus process in the session that ensures all clients have same data

Reply:
I agree about writing this setting to a file. We can follow these steps:
1. Write the timeout to the summary during creation and then use this same value for the life time of the document.
2. Ship the feature dark, i.e., the timeout is undefined for now.
3. Enable the feature at some point in future. All containers created after this point in time will have the timeout and will all be on the same page.

For testing, should be add a local storage flag to enable this feature?

Reply:
It's fine to expose overwrites to host, and host using these overwrites in dev environment to get feedback,
But I think enabling these overwrites means newly created files (in dev environment) will have certain capabilities. It will impact production of course (as these same files will be opened in prod), but it will impact only us (some internal MS rings) as ideally there should be close to zero files we create in dev environment that we share with external world (and I think cross-tenant sharing does not work anyway).

So yes, it's fine to have local storage flags, but they should not impact existing files. We can test just fine whole process. The only caveat - clearly, we do not want 30 days for internal testing. Even 2 days (min bar to snapshot caching limit) is too long. So, we should think how we short-certain that process. Maybe we do have local storage flags that just screw up all these settings, but only few people should know and use these, on very small set of files that you know nobody else will open.